src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/tasks/MigratedCaseLinkUpdatesTask.java

```diff
@@ -0,0 +1,196 @@
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ecm.common.client.CcdClient;
+import uk.gov.hmcts.et.common.model.ccd.CCDRequest;
+import uk.gov.hmcts.et.common.model.ccd.CaseDetails;
+import uk.gov.hmcts.et.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.AdminUserService;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.CaseRetrievalForCaseWorkerService;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.FeatureToggleService;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.exceptions.CaseDuplicateSearchException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.ACCEPTED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLOSED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.REJECTED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SUBMITTED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.TRANSFERRED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.VETTED_STATE;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class MigratedCaseLinkUpdatesTask {
+    private final AdminUserService adminUserService;
+    private final CaseRetrievalForCaseWorkerService caseRetrievalForCaseWorkerService;
+    private final CcdClient ccdClient;
+    private final FeatureToggleService featureToggleService;
+    private final List<String> validStates = List.of(TRANSFERRED_STATE, ACCEPTED_STATE, REJECTED_STATE,
+            SUBMITTED_STATE, VETTED_STATE, CLOSED_STATE);
+    private static final String EVENT_ID = "migrateCaseLinkDetails";
+    @Value("${cron.caseTypeId}")
+    private String caseTypeIdsString;
+
+    @Value("${cron.maxCasesPerSearch}")
+    private int maxCases;
+    private static final int TWO = 2;
+
+    @Scheduled(cron = "${cron.updateTransferredCaseLinks}")
+    public void updateTransferredCaseLinks() {
+
+        if (!featureToggleService.isUpdateTransferredCaseLinksEnabled()) {
+            return;
+        }
+
+        String query = buildStartQuery();
+        String adminUserToken = adminUserService.getAdminUserToken();
+        String[] caseTypeIds = caseTypeIdsString.split(",");
+
+        Arrays.stream(caseTypeIds).forEach(caseTypeId -> {
+            try {
+                //Get transferred cases by case type
+                List<SubmitEvent> transferredCases = ccdClient.buildAndGetElasticSearchRequest(adminUserToken,
+                        caseTypeId, query);
+
+                for (SubmitEvent submitEvent : transferredCases) {
+                    //find possible duplicate cases by ethos reference
+                    //list of pairs of 'case type id' and 'list of submit events'
+                    List<Pair<String, List<SubmitEvent>>> listOfPairs = findCaseByEthosReference(
```

Comment:
In case you need to stop the CRON in the middle of a run
```suggestion
                     if (!featureToggleService.isUpdateTransferredCaseLinksEnabled()) {
                         return;
                     }
                    //find possible duplicate cases by ethos reference
                    //list of pairs of 'case type id' and 'list of submit events'
                    List<Pair<String, List<SubmitEvent>>> listOfPairs = findCaseByEthosReference(
```